kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/expressions/types/dtable/AddInputClauseCommand.java

```
@@ -85,8 +87,10 @@ public AddInputClauseCommand(final DecisionTable dtable,
             @Override
             public CommandResult<RuleViolation> undo(final GraphCommandExecutionContext context) {
                 final int clauseIndex = dtable.getInput().indexOf(inputClause);
-                dtable.getRule().forEach(rule -> rule.getInputEntry().remove(clauseIndex));
-                dtable.getInput().remove(inputClause);
+                if (clauseIndex != -1) {
```

If the ```InputClause``` could not be found (i.e. ```index == -1```) something quite catastrophic has happened and I'd rather throw an exception than silently ignore (as this now does).. In such eventualities it would point to a bug somewhere else that this simply hides... so IMO I'd simply prefer the ```remove(-1)``` to fail and throw an exception to Users or if we keep your check we throw a more "user friendly" message.